### PR TITLE
[Flax] Correctly Add MT5

### PR DESCRIPTION
**Pull request by patrickvonplaten**
# What does this PR do?

During the Flax sprint many teams weren't aware that mt5 models can be used with the `FlaxT5ForConditionalGeneration` class. This is mainly because the docs currently state that FlaxMT5 is not implemented: https://huggingface.co/transformers/index.html#supported-frameworks and because there are no docs on FlaxMT5, but for PyTorch & TF (https://huggingface.co/transformers/model_doc/mt5.html).

This PR adds a FlaxMT5 class analog to PT and TF and also adds official Flax weights to `mt5-base`, etc.: https://huggingface.co/google/mt5-base/commit/0b908f9e3c2fabccc4ab411b89838cecdd9ad499

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

File: utils/check_repo.py [hunk 1]
```diff
@@ -82,8 +82,7 @@
 # trigger the common tests.
 TEST_FILES_WITH_NO_COMMON_TESTS = [
     "test_modeling_camembert.py",
-    "test_modeling_flax_bert.py",
```
Comment (patrickvonplaten) [inline utils/check_repo.py:85]: They are actually correctly implemented

File: tests/test_modeling_flax_mt5.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
+# Copyright 2020 The HuggingFace Team. All rights reserved.
```
Comment (patil-suraj) [inline tests/test_modeling_flax_mt5.py:1]:
```suggestion
# Copyright 2021 The HuggingFace Team. All rights reserved.
```

**Review by patil-suraj — APPROVED** [review]
Thanks for adding this!

**Review by sgugger — COMMENTED** [review]
Do we really need to add an extra class? Can't we just have the auto-mapping point to a FlaxT5Model?

**Comment by patrickvonplaten** [comment]
> Do we really need to add an extra class? Can't we just have the auto-mapping point to a FlaxT5Model?

Technically we don't need it, my main arguments are:

- Consistency with PyTorch & TF (people thought MT5 can't be used with Flax because https://huggingface.co/transformers/model_doc/mt5.html doesn't have Flax classes)
- Ability to provide meaningful examples for MT5 in Flax. T5 is not multi-lingual so the examples might be misleading for Flax

File: utils/check_repo.py (hunk 1 above)
Comment (sgugger) [reply utils/check_repo.py:85]: This is from a long time ago ;-)

**Review by sgugger — APPROVED** [review]
Good for me then, thanks for adding this model!

**Pull request merged**